[exporter/clickhouse] create database fail

### Component(s)

exporter/clickhouse

### What happened?

## Description
OTel collector can't start with clickhouse exporter when the database doesn't exist.  error log:
```
Error: cannot start pipelines: create database:code: 81, message: Database otel doesn't exist; failed to shutdown pipelines: no existing monitoring routine is running; no existing monitoring routine is running
2023/06/25 17:25:35 collector server run finished with error: cannot start pipelines: create database:code: 81, message: Database otel doesn't exist; failed to shutdown pipelines: no existing monitoring routine is running; no existing monitoring routine is running
```

I found that 
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/da5df85e77bb717e87096c20663ddc8bbb5ebf8a/exporter/clickhouseexporter/exporter_logs.go#L181-L200
The `createDatabase()` needs `buildDB()` to replace the origin database to `defaultDatabase`(e.g. `tcp://127.0.0.1:9000/otel` to `tcp://127.0.0.1:9000/default`) however `cfg.buildDB` doesn't cover that.


### Collector version

latest

### Environment information

_No response_

### OpenTelemetry Collector configuration

```yaml
receivers:
  fluentforward:
    endpoint: 0.0.0.0:24224
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
processors:
  batch:
    send_batch_size: 100000
    timeout: 5s
  memory_limiter:
    check_interval: 2s
    limit_mib: 1800
    spike_limit_mib: 500
  resourcedetection/system:
    detectors: [ "system" ]
    system:
      hostname_sources: [ "os" ]
  resource:
    attributes:
      - key: service.name
        value: "serviceName"
        action: upsert
exporters:
  clickhouse:
    endpoint: tcp://172.0.0.1:9000
    database: otel
    logs_table_name: otel_logs
    traces_table_name: otel_traces
    ttl_days: 3
    timeout: 10s
    sending_queue:
      queue_size: 100
    retry_on_failure:
      enabled: true
      initial_interval: 5s
      max_interval: 30s
      max_elapsed_time: 300s
extensions:
  health_check:
  pprof:
  zpages:
  memory_ballast:
    size_mib: 1000

service:
  extensions: [ pprof, zpages, health_check ]
  pipelines:
    logs:
      receivers: [ fluentforward, otlp ]
      processors: [ memory_limiter, resourcedetection/system, resource, batch ]
      exporters: [ clickhouse ]
    traces:
      receivers: [ otlp ]
      processors: [ memory_limiter, resourcedetection/system, resource, batch ]
      exporters: [ clickhouse ]
```


### Log output

```shell
Error: cannot start pipelines: create database:code: 81, message: Database otel doesn't exist; failed to shutdown pipelines: no existing monitoring routine is running; no existing monitoring routine is running
2023/06/25 17:25:35 collector server run finished with error: cannot start pipelines: create database:code: 81, message: Database otel doesn't exist; failed to shutdown pipelines: no existing monitoring routine is running; no existing monitoring routine is running
```


### Additional context

_No response_

I use this [ut](https://github.com/Frapschen/opentelemetry-collector-contrib/blob/fix_create_database_fail/exporter/clickhouseexporter/config_test.go#L230) to reproduce it. 
```
=== RUN   TestConfig_buildDSN
=== RUN   TestConfig_buildDSN/Should_support_use_default_database_to_create_new_database
    config_test.go:263: 
        	Error Trace:	/Users/fraps/daocloud/github-code/opentelemetry-collector-contrib/exporter/clickhouseexporter/config_test.go:263
        	Error:      	Not equal: 
        	            	expected: "tcp://127.0.0.1:9000/default"
        	            	actual  : "tcp://127.0.0.1:9000/otel"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-tcp://127.0.0.1:9000/default
        	            	+tcp://127.0.0.1:9000/otel
        	Test:       	TestConfig_buildDSN/Should_support_use_default_database_to_create_new_database
        	Messages:   	buildDSN(default)
--- FAIL: TestConfig_buildDSN (0.00s)
    --- FAIL: TestConfig_buildDSN/Should_support_use_default_database_to_create_new_database (0.00s)
```

--- a/.chloggen/fix-create-database-fail.yaml
+++ b/.chloggen/fix-create-database-fail.yaml
@@ -1,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: clickhouseexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix clickhouse exporter create database fail
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [23664]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/exporter/clickhouseexporter/config.go
+++ b/exporter/clickhouseexporter/config.go
@@ -103,8 +103,15 @@ func (cfg *Config) buildDSN(database string) (string, error) {
 	// Override database if specified in config.
 	if cfg.Database != "" {
 		dsnURL.Path = cfg.Database
-	} else if database == "" && cfg.Database == "" && dsnURL.Path == "" {
-		// Use default database if not specified in any other place.
+	}
+
+	// Override database if specified in database param.
+	if database != "" {
+		dsnURL.Path = database
+	}
+
+	// Use default database if not specified in any other place.
+	if database == "" && cfg.Database == "" && dsnURL.Path == "" {
 		dsnURL.Path = defaultDatabase
 	}
 

--- a/exporter/clickhouseexporter/config_test.go
+++ b/exporter/clickhouseexporter/config_test.go
@@ -147,7 +147,7 @@ func TestConfig_buildDSN(t *testing.T) {
 				Database: "otel",
 			},
 			args: args{
-				database: defaultDatabase,
+				database: "otel",
 			},
 			wantChOptions: ChOptions{
 				Secure: false,
@@ -201,7 +201,8 @@ func TestConfig_buildDSN(t *testing.T) {
 			},
 			args: args{},
 			want: "clickhouse://127.0.0.1:9000/default?foo=bar&secure=true",
-		}, {
+		},
+		{
 			name: "Parse clickhouse settings",
 			fields: fields{
 				Endpoint: "https://127.0.0.1:9000?secure=true&dial_timeout=30s&compress=lz4",
@@ -225,6 +226,16 @@ func TestConfig_buildDSN(t *testing.T) {
 			},
 			args: args{},
 			want: "clickhouse://127.0.0.1:9000/default?foo=bar&secure=true",
+		},
+		{
+			name: "support replace database in DSN to default database",
+			fields: fields{
+				Endpoint: "tcp://127.0.0.1:9000/otel",
+			},
+			args: args{
+				database: defaultDatabase,
+			},
+			want: "tcp://127.0.0.1:9000/default",
 		},
 	}
 	for _, tt := range tests {

